src/modules/directory.rs

```diff
@@ -101,169 +121,862 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-/// Contract the root component of a path
-///
-/// Replaces the `top_level_path` in a given `full_path` with the provided
-/// `top_level_replacement`.
-fn contract_path(full_path: &Path, top_level_path: &Path, top_level_replacement: &str) -> String {
-    if !full_path.starts_with(top_level_path) {
-        return full_path.to_slash().unwrap();
-    }
-
-    if full_path == top_level_path {
-        return top_level_replacement.to_string();
-    }
-
-    format!(
-        "{replacement}{separator}{path}",
-        replacement = top_level_replacement,
-        separator = "/",
-        path = full_path
-            .strip_prefix(top_level_path)
-            .unwrap()
-            .to_slash()
-            .unwrap()
-    )
+#[derive(PartialEq, Clone, Debug)]
+pub enum DirectoryComponent {
+    Prefix(String),
+    Root,
+    Current,
+    Parent,
+    Home,
```

Comment:
Would `Normal(options.home)` work instead of `Home`?

Reply:
Probably but keeping it as a string would only confuse when replacing with the configured symbol is the only use for it.
This way, we don't have to check if this "Normal" is a home.
Also, we could evolve it into Home(String) if we want to support ~user/folder which Normal(options.Home) won't support.